x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentTokenForInitialNode.java

```diff
@@ -0,0 +1,187 @@
+            environment -> new EnrollmentTokenGenerator(environment)
+        );
+    }
+
+    BootstrapPasswordAndEnrollmentTokenForInitialNode(Function<Environment, CommandLineHttpClient> clientFunction,
+                                                      CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction,
+                                                      CheckedFunction<Environment, EnrollmentTokenGenerator, Exception>
+                                                     createEnrollmentTokenFunction){
+        super("Set elastic password and generate enrollment token for initial node");
+        this.clientFunction = clientFunction;
+        this.keyStoreFunction = keyStoreFunction;
+        this.createEnrollmentTokenFunction = createEnrollmentTokenFunction;
+        parser.allowsUnrecognizedOptions();
+        keyStorePassword = parser.accepts("keystore-password", "keystore password").withRequiredArg();
+        docker = parser.accepts("docker", "determine that we are running in docker");
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new BootstrapPasswordAndEnrollmentTokenForInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (!options.has(keyStorePassword)) {
+            throw new UserException(ExitCodes.USAGE, null);
+        }
+
+        final SecureString keystorePassword = new SecureString(keyStorePassword.values(options).get(0));
+        SecureString password;
+        final CommandLineHttpClient client = getClient(env);
+        final EnrollmentTokenGenerator enrollmentTokenGenerator = getEnrollmentTokenGenerator(env);
+        keyStoreWrapper = getKeyStoreWrapper(env);
+        final SecureString bootstrapPassword = readBootstrapPassword(env, keystorePassword);
+        try {
+            client.checkClusterHealthWithRetriesWaitingForCluster(elasticUser, bootstrapPassword, 5);
+        } catch (Exception e) {
+            throw new UserException(ExitCodes.UNAVAILABLE, null);
+        }
+        if (ReservedRealm.BOOTSTRAP_ELASTIC_PASSWORD.exists(env.settings())) {
+            password = bootstrapPassword;
+        } else {
+            password = setElasticUserPassword(client, bootstrapPassword);
+            terminal.println("elastic user password: " + password);
+        }
+        final EnrollmentToken kibanaToken;
+        try {
+            kibanaToken = enrollmentTokenGenerator.createKibanaEnrollmentToken(elasticUser, password);
+        } catch (Exception e) {
+            throw new UserException(ExitCodes.UNAVAILABLE, null);
+        }
+        terminal.println("CA fingerprint: " + kibanaToken.getFingerprint());
+        terminal.println("Kibana enrollment token: " + kibanaToken.encode());
+        if (options.has(docker)) {
+            final EnrollmentToken nodeToken;
```

Comment:
nit, you can move the println in the try clause so that you don't have to split the declaration above